app/code/Magento/InventoryCatalog/Model/ResourceModel/BulkInventoryTransfer.php

```diff
@@ -205,7 +205,7 @@ public function execute(
         $connection->beginTransaction();
         foreach ($types as $sku => $type) {
```

Comment:
btw why do we have this foreach here which leads to execution of SQL query for each item in the array.

but not collect all items which satisfy the condition `isSourceItemManagementAllowedForProductType ` and process them all together with Bulk transferInventory